components/raftstore/src/store/peer.rs

```diff
@@ -4791,7 +4791,10 @@ where
             }
         }
         let flags = WriteBatchFlags::from_bits_check(req.get_header().get_flags());
-        if flags.contains(WriteBatchFlags::STALE_READ) {
+        // Leader has the latest data, so we can skip safe_ts check, otherwise
```

Comment:
It may not have latest data if it just becomes leader.

Reply:
Even if it has latest data, who to do with locks? For example, if it reads using ts1 while some transaction is being committed using ts1 - 1, how to handle the race?

Reply:
`KeyIsLocked` will be returned to TiDB. So I suppose TiDB or user will handle the race.

Reply:
Can you verify that? As if it's a stale read and read ts is less than safe ts, then all locks can just be ignored.

Reply:
> As if it's a stale read and read ts is less than safe ts

Wait... in this case, it's safe to read even if in a follower.

Reply:
Yes, as long as *read ts <= safe ts is checked".